app/src/main/java/com/kickstarter/ui/activities/CommentsActivity.kt

```diff
@@ -0,0 +1,195 @@
+import com.jakewharton.rxbinding.view.RxView
+import com.jakewharton.rxbinding.widget.RxTextView
+import com.kickstarter.R
+import com.kickstarter.databinding.CommentsLayoutBinding
+import com.kickstarter.libs.ActivityRequestCodes
+import com.kickstarter.libs.BaseActivity
+import com.kickstarter.libs.RecyclerViewPaginator
+import com.kickstarter.libs.SwipeRefresher
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.ObjectUtils
+import com.kickstarter.libs.utils.TransitionUtils
+import com.kickstarter.libs.utils.ViewUtils
+import com.kickstarter.models.Project
+import com.kickstarter.ui.IntentKey
+import com.kickstarter.ui.adapters.CommentsAdapter
+import com.kickstarter.ui.data.LoginReason
+import com.kickstarter.ui.viewholders.EmptyCommentsViewHolder
+import com.kickstarter.ui.viewholders.ProjectContextViewHolder
+import com.kickstarter.viewmodels.CommentsViewModel
+import com.trello.rxlifecycle.ActivityEvent
+import rx.Observable
+import rx.android.schedulers.AndroidSchedulers
+import rx.subjects.PublishSubject
+
+@RequiresActivityViewModel(CommentsViewModel.ViewModel::class)
+class CommentsActivity : BaseActivity<CommentsViewModel.ViewModel>(), CommentsAdapter.Delegate {
+    private var adapter: CommentsAdapter? = null
+    private var recyclerViewPaginator: RecyclerViewPaginator? = null
+    private var swipeRefresher: SwipeRefresher? = null
+    private val alertDialog = PublishSubject.create<AlertDialog>()
+    private lateinit var binding: CommentsLayoutBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = CommentsLayoutBinding.inflate(layoutInflater)
+        val view: View = binding.root
+        setContentView(view)
+        adapter = CommentsAdapter(this)
+        binding.commentsRecyclerView.adapter = adapter
+        binding.commentsRecyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerViewPaginator = RecyclerViewPaginator(binding.commentsRecyclerView, { viewModel.inputs.nextPage() }, viewModel.outputs.isFetchingComments)
+        swipeRefresher = SwipeRefresher(
+            this, binding.commentsSwipeRefreshLayout, { viewModel.inputs.refresh() }
+        ) { viewModel.outputs.isFetchingComments }
+        val commentBodyEditText = alertDialog
+            .map { it?.findViewById<TextView>(R.id.comment_body) }
+        val postCommentButton = alertDialog
+            .map { it?.findViewById<TextView>(R.id.post_button) }
+        val cancelButton = alertDialog
+            .map { it?.findViewById<TextView>(R.id.cancel_button) }
+        cancelButton
+            .switchMap { view -> view?.let { RxView.clicks(it) } }
+            .observeOn(AndroidSchedulers.mainThread())
+            .compose(bindToLifecycle())
+            .subscribe { viewModel.inputs.commentDialogDismissed() }
```

Comment:
nit picky detail, can we let a line separation between each subscription? it's hard to read without separations

Reply:
+1